monitor.cpp

```diff
@@ -1,38 +1,51 @@
 #include "./monitor.h"
-#include <assert.h>
+#include <iostream>
 #include <thread>
 #include <chrono>
-#include <iostream>
+
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
+bool isTemperatureOk(float temperature) {
+    return temperature >= 95 && temperature <= 102;
+}
+
+bool isPulseRateOk(float pulseRate) {
+    return pulseRate >= 60 && pulseRate <= 100;
+}
+
+bool isSpO2Ok(float spo2) {
+    return spo2 >= 90;
+}
+
+VitalStatus evaluateVitals(float temperature, float pulseRate, float spo2) {
+    if (!isTemperatureOk(temperature)) return VitalStatus::TemperatureError;
+    if (!isPulseRateOk(pulseRate)) return VitalStatus::PulseError;
+    if (!isSpO2Ok(spo2)) return VitalStatus::SpO2Error;
```

Comment:
Nice change to return the specific problem, instead of just a boolean. Naturally, the customer wants to see what's wrong with the vitals, beyond seeing that something is 'not ok'.